infrastructure/shifty.bicep

```diff
@@ -0,0 +1,41 @@
+param location string = resourceGroup().location
+
+param environment string
+
+param targetBranch string
+
+param organizationPrefix string
+param applicationPrefix string
+
+param sharedResourceGroupName string
+
+resource staticwebapp 'Microsoft.Web/staticSites@2022-03-01' = {
+  name: 'stapp-${organizationPrefix}-${applicationPrefix}-${environment}'
+  location: location
+  sku: {
+    name: 'Free'
+    tier: 'Free'
+  }
+  properties: {
+    allowConfigFileUpdates: false
+    repositoryUrl: 'https://github.com/AnalogIO/shifty-webapp'
+    branch: targetBranch
+    provider: 'GitHub'
+    stagingEnvironmentPolicy: 'Disabled'
+    enterpriseGradeCdnStatus: 'Disabled'
+  }
+}
+
+module dns 'modules/dns.bicep' = {
+  name: '${deployment().name}-dns'
+  scope: resourceGroup(sharedResourceGroupName)
+  params: {
+    environment: environment
+    webappAzureGeneratedFqdn: staticwebapp.properties.defaultHostname
+  }
+}
+
+resource staticwebappCustomDomain 'Microsoft.Web/staticSites/customDomains@2022-03-01' = {
```

Comment:
It would be cool if we could take the full FQDN and export it from the DNS module such that we could rely on the value from there instead of doing a string concatenation here

Reply:
I thought this too, but it Bicep complains that the "name" key needs to be a value that can be calculated at the start of deployment, which i guess is not the case for the exported FQDN from the DNS module